packages/core/src/component/state.ts

```diff
@@ -0,0 +1,52 @@
+import { createObservable, ObjectType, Observable } from '../utils';
+
+export interface StateValue<TModel, TDictionary> {
+  model: TModel;
+  dictionary: TDictionary;
+}
+
+export interface State<TModel, TDictionary>
+  extends Observable<StateValue<TModel, TDictionary>>{
+  getValue: () => StateValue<TModel, TDictionary>;
+  addUpdateChunk: (statePart: Partial<StateValue<Partial<TModel>, Partial<TDictionary>>>) => void;
+  commitUpdates: (rollback?: boolean) => void;
+}
+
+export const createState = <TModel extends ObjectType, TDictionary extends ObjectType>(
+  initialState: StateValue<TModel, TDictionary>,
+): State<TModel, TDictionary> => {
+  let stateValue = initialState;
+  let updatedStateValue = initialState;
```

Comment:
Using `value` in this way is very confusing for me.  When I see a `value` alongside a `state` I consider the first one as a part of the second.

```suggestion
  let state = initialState;
  let nextState = initialState;
```

Reply:
We decided to rename these variables to current/next.